cmake/BLTMacros.cmake

```diff
@@ -663,3 +663,56 @@ macro(blt_append_custom_compiler_flag)
 
 endmacro(blt_append_custom_compiler_flag)
 
+
+
+
+##------------------------------------------------------------------------------
+## blt_find_libraries( <FOUND_LIBS>
+##                     NAMES [libname1 [libname2 ...]]
+##                     PATHS [path1 [path2 ...]] )
+##
+## This command is used to find a list of libraries. A cache entry named by <FOUND_LIBS> 
+## is created to store the result of this command. If the libraries are found the
+## results are stored in FOUND_LIBS.
+##
+## NAMES lists the names of the libraries to search for.
+##
+## PATH lists the paths in which to search for NAMES.
+##
+##
+##------------------------------------------------------------------------------
+macro(blt_find_libraries FOUND_LIBS)
+
+    set(multiValueArgs NAMES
```

Comment:
Does it make sense to have a set of required ``NAMES`` and a set of ``OPTIONAL_NAMES``?  The way this is written, if the library is not found, the configuration will fail.

Alternatively, does it make sense to add a parameter to make the results optional, similar to how the ``REQUIRED`` keyword works in cmake`s ``find(FOO REQUIRED)``?

Reply:
A different useful feature, somewhat in line with @white238's request for foo and libfoo would be if you want to find at least one library in the list, 

A (possible bad) example would be if you want to find ``foo`` or ``foo.so.6``.

Reply:
I agree an optional list would be very useful.  I would on the other hand not do a required keyword.

Reply:
I can do this, but I am wondering what would be the advantage of this option? For instance, I am using this to find specific libraries, and if they are not found then cmake gives an error. If they are found, things link fine. If there is an optional library you won't know you missed it until you get a link error? I would think this is worse? Perhaps I am missing something here?

Reply:
I changed NAMES to REQUIRED_NAMES and OPTIONAL_NAMES. Failure to find any of the REQUIRED_NAMES results in a FATAL_ERROR. Failure to find any of the OPTIONAL_NAMES prints a warning.